salt/modules/cmdmod.py

```diff
@@ -245,6 +245,10 @@ def _run(cmd,
     '''
     Do the DRY thing and only call subprocess.Popen() once
     '''
+    # Remove whitespace from strings
+    if isinstance(cmd, six.string_types):
+        cmd = cmd.strip()
+
```

Comment:
It wasn't a split... it was a strip...

Anyway, I just pushed a more complicated... solution. Let me know what you think.